### ISSUE TITLE
`turbo` should accept folders without globs `/**` in `pipeline` `outputs`

### ISSUE THREAD

**Issue by jaredpalmer** [vercel/turbo#1048]
This initially bit @Rich-Harris when moving SvelteKit to `turbo`. Either we should warn against this or allow it. At the moment, we fail silently

```jsonc
{
  "pipeline": {
       "build": {
          "outputs": [".next"] // this fails silently now, but should be allowed 
       }
  }
}
```

Area of discussion: should we treat folders and globs the same? Are there perf gains to be had by moving away from globs?

**Comment by nathanhammond** [comment]
We need to do a few things as part of this:

- [ ] Come up with a testing strategy. (Probably adopt afero?)
- [ ] Write tests to constrain and define the current behavior.
- [ ] Ship the incremental change.

It's likely we can reason about the patch fix for this and not screw it up but we should be conservative on changing how we specify what is going into the cache. Being wrong here has significant consequences we should be careful with.

**Comment by Leksat** [comment]
We could use folders without globs to solve https://github.com/vercel/turbo/issues/4566

**Comment by neongreen** [comment]
> Fails silently

This is pretty bad! I only found this issue by accident

**Comment by anthonyshew** [comment]
This is fixed in Turborepo 2.0: https://github.com/vercel/turbo/pull/8180
